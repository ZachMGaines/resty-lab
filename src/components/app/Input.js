import React from 'react';

import PropTypes from 'prop-types';


const Input = ({ url, method, putBody, onUrlChange,
  onSubmit, onMethodChange, onPutChange }) => {
  return <form className="actualForm" onSubmit={onSubmit}>
    <input placeholder="URL" className="searchBar" value={url}
      onChange={onUrlChange}></input>
    <section className="options">
      <input id="get" type="radio" name="method" value="GET"
        method={method} onChange={onMethodChange}></input>
      <label htmlFor="get">GET</label>
      <input id="post" type="radio" name="method" value="POST"
        method={method} onChange={onMethodChange}></input>
      <label htmlFor="post">POST</label>
      <input id="put" type="radio" name="method" value="PUT"
        method={method} onChange={onMethodChange}></input>
      <label htmlFor="put">PUT</label>
      <input id="patch" type="radio" name="method" value="PATCH"
        method={method} onChange={onMethodChange}></input>
      <label htmlFor="patch">PATCH</label>
      <input id="delete" type="radio" name="method" value="DELETE"
        method={method} onChange={onMethodChange}></input>
      <label htmlFor="delete">DELETE</label>
      <button>Go!</button>
    </section>
    <textarea placeholder="Raw JSON Body" name="body"
      value={putBody} onChange={onPutChange}></textarea>
  </form>;
};

Input.propTypes = {
  url: PropTypes.string,
  method: PropTypes.string,
  putBody: PropTypes.string,
  onUrlChange: PropTypes.func,
  onSubmit: PropTypes.func,
  onMethodChange: PropTypes.func,
  onPutChange: PropTypes.func
};

export default Input;
