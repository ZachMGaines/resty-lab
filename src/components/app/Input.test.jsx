import React from 'react';
import { render } from '@testing-library/react';
import Input from './Input';

describe('Input component', () => {
  it('renders inputArea', () => {
    const { asFragment } = render(
      <Input
        url="aintitcool.com"
        method="put"
        putBody="name: dexter"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
