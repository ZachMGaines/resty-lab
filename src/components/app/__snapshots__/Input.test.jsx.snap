// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders inputArea 1`] = `
<DocumentFragment>
  <form
    class="actualForm"
  >
    <input
      class="searchBar"
      placeholder="URL"
      value="aintitcool.com"
    />
    <section
      class="options"
    >
      <input
        id="get"
        method="put"
        name="method"
        type="radio"
        value="GET"
      />
      <label
        for="get"
      >
        GET
      </label>
      <input
        id="post"
        method="put"
        name="method"
        type="radio"
        value="POST"
      />
      <label
        for="post"
      >
        POST
      </label>
      <input
        id="put"
        method="put"
        name="method"
        type="radio"
        value="PUT"
      />
      <label
        for="put"
      >
        PUT
      </label>
      <input
        id="patch"
        method="put"
        name="method"
        type="radio"
        value="PATCH"
      />
      <label
        for="patch"
      >
        PATCH
      </label>
      <input
        id="delete"
        method="put"
        name="method"
        type="radio"
        value="DELETE"
      />
      <label
        for="delete"
      >
        DELETE
      </label>
      <button>
        Go!
      </button>
    </section>
    <textarea
      name="body"
      placeholder="Raw JSON Body"
    >
      name: dexter
    </textarea>
  </form>
</DocumentFragment>
`;
